tutorxqueue/plugin.py

```diff
@@ -233,19 +259,38 @@ def grade_submission(self, submission_id, submission_key, grade, correct, msg):
             },
         )
 
-    def request(self, endpoint, method="GET", data=None, params=None):
+    def request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        data: Optional[Dict[str, Any]] = None,
+        params: Optional[Dict[str, Any]] = None,
+    ) -> Any:
         func = getattr(self.session, method.lower())
-        response = func(self.url(endpoint), data=data, params=params)
-        # TODO handle errors >= 400 and non-parsable json responses
-        return response.json()
+        try:
+            response: requests.Response = func(
+                self.url(endpoint), data=data, params=params
+            )
+            response.raise_for_status()  # This will raise a HTTPError if the response was an unsuccessful one
+        except requests.HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+            return None
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+            return None
+        else:
+            try:
+                return response.json()
+            except ValueError:
+                print("Could not parse json")
+            return None
```

Comment:
While I understand that you are trying to handle errors properly, I'd rather keep this in a separate PR for now. That's because error handling can be a tricky affair...

Reply:
Agree. I'll take care of that.